[gdal] build failure

**Host Environment**
- OS: windows
- Compiler: MSVC/14.32.31326 (i.e. VS 2022)

**To Reproduce**
Steps to reproduce the behavior: 
`./vcpkg.exe install gdal --triplet x64-windows`

**Failure logs**
`-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: "C:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v
    Working Directory: C:/build/vcpkg/buildtrees/gdal/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\build\vcpkg\buildtrees\gdal\config-x64-windows-CMakeCache.txt.log
      C:\build\vcpkg\buildtrees\gdal\config-x64-windows-CMakeCache.txt.log
      C:\build\vcpkg\buildtrees\gdal\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:238 (vcpkg_execute_required_process)
  ports/gdal/portfile.cmake:62 (vcpkg_cmake_configure)
  scripts/ports.cmake:146 (include)


error: building gdal:x64-windows failed with: BUILD_FAILED`


**Additional context**
Builds with default-features and hda5 removed, i.e. gdal[core,curl,geos,libspatialite,netcdf,postgresql,recommended-features]
CMake is 3.24.0-rc3

[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/9071293/config-x64-windows-out.log)
[config-x64-windows-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/9071294/config-x64-windows-CMakeCache.txt.log)

--- a/ports/gdal/vcpkg.json
+++ b/ports/gdal/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "gdal",
   "version-semver": "3.5.1",
+  "port-version": 1,
   "description": "The Geographic Data Abstraction Library for reading and writing geospatial raster and vector data",
   "homepage": "https://gdal.org",
   "license": null,
@@ -115,7 +116,10 @@
       "dependencies": [
         {
           "name": "hdf5",
-          "default-features": false
+          "default-features": false,
+          "features": [
+            "cpp"
+          ]
         }
       ]
     },

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2466,7 +2466,7 @@
     },
     "gdal": {
       "baseline": "3.5.1",
-      "port-version": 0
+      "port-version": 1
     },
     "gdcm": {
       "baseline": "3.0.12",

--- a/versions/g-/gdal.json
+++ b/versions/g-/gdal.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "dd8ff4159201d96aecb0cc2de325b56a522ae0e1",
+      "version-semver": "3.5.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "f623db2a0771ad4df6c248b046cd8f369ecbdcca",
       "version-semver": "3.5.1",
       "port-version": 0

